docs/search.md

```diff
@@ -0,0 +1,45 @@
+Search
+======
+
+The search endpoint provides a powerful way of finding items that are accessible by a single user or an entire 
+enterprise. Leverage the parameters listed below to generate targeted advanced searches.
+
+Search for Content
+------------------
+
+To get a list of items matching a serch query, call [`search.query(query, limit=None, offset=0, **kwargs)`][query] will return an `Iterable` that allows you 
+to iterate over the [`Item`][item_class] objects in the collection.
+
+```python
+items = client.search().query(query='TEST QUERY', limit=100, file_extensions=['pdf', 'doc'])
+for item in items:
+    print('The item ID is {} and the item name is {1}'.format(item.id, item.name)
+```
+
+[query]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.search.Search.query
+[item_class]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.item.Item
+
+### Metadata Search
+
+To filter by metadata in your search, first create [MetadataSearchFilter][metadata_search_filter] object with the 
+specified `template_key` and `scope` as well as adding filter, `field_key` and `value` with 
+[`metadata_search_filter.add_value_based_filter(field_key, value)`][add_value_based_filter]. Next, create a 
```

Comment:
Consider shortening the object name here to `metadata_filter` to align with the next line where it's passed into `add_filter()`